src/library_pthread.js

```diff
@@ -465,6 +465,20 @@ var LibraryPThread = {
 
     getNewWorker: function() {
       if (PThread.unusedWorkers.length == 0) {
+#if !PROXY_TO_PTHREAD
+#if ASSERTIONS || PTHREADS_DEBUG
+        err('Tried to spawn a new thread, but the thread pool is exhausted.\n' +
+        'This will result in a deadlock unless the code explicitly breaks out to the event loop.\n' +
+        'If you want to increase the pool size, use setting `-s PTHREAD_POOL_SIZE=...`.'
+#if !PTHREAD_POOL_SIZE_STRICT
+        + '\nIf you want to throw an explicit error instead of deadlocking in those cases, use setting `-s PTHREAD_POOL_SIZE_STRICT=1`.'
```

Comment:
I don't think the text here is precise. There are cases in which a deadlock would not occur, but that option would cause an error to be thrown. Perhaps it could say "instead of the risk of deadlocking"?

Reply:
It says "unless the code explicitly breaks out to the event loop" above - is that not enough?

Reply:
I suppose another case where it won't fail is if threads don't wait for each other, and can finish & be returned to the pool and eventually reused by the "waiting" ones.

I'll reword a bit.